[MC/DC] Make covmap tolerant of nested Decisions

chapuni opened the pull request:
CoverageMappingWriter reorders `Region`s by `endLoc DESC` to prioritize wider `Decision` with the same `startLoc`.

In `llvm-cov`, tweak seeking Decisions by reversal order to find smaller Decision first.

chapuni committed:
Merge branch 'users/chapuni/cov/single/test' into users/chapuni/cov/single/merge

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts.cpp

chapuni committed:
Update tests

chapuni committed:
Merge branches 'users/chapuni/cov/single/merge' and 'users/chapuni/cov/binary' into users/chapuni/cov/single/refactor-base

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test
	llvm/test/tools/llvm-cov/showLineExecutionCounts.test

chapuni committed:
New SingleByteCoverage

chapuni committed:
Merge branch 'users/chapuni/cov/single/refactor' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branches 'users/chapuni/cov/single/loop', 'users/chapuni/cov/single/condop' and 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
threads.c => threads.test (following #113114)

chapuni committed:
clang/test: Add tests that we missed while I was updating.

chapuni committed:
Introduce CounterMappingRegion::isBranch(). NFC.

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp
	llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h

chapuni committed:
Merge branches 'users/chapuni/cov/single/test' and 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Update the test

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Update the test

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Update desc

chapuni committed:
s/Count1/BinaryCount/

chapuni committed:
Merge branch 'main' into users/chapuni/cov/binary

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test
	llvm/test/tools/llvm-cov/showLineExecutionCounts.test

chapuni committed:
Merge branch 'users/chapuni/cov/binary' into users/chapuni/cov/single/refactor

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/getpair

chapuni committed:
Introduce BranchCounterPair

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/pair

chapuni committed:
void verifyCounterMap() const

chapuni committed:
Introduce the dedicated class CounterPair instead of std::pair

chapuni committed:
Merge branches 'users/chapuni/cov/single/getpair' and 'users/chapuni/cov/single/pair' into users/chapuni/cov/single/nextcount-base

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount-base' into users/chapuni/cov/single/nextcount

Conflicts:
	clang/lib/CodeGen/CodeGenPGO.cpp
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Introduce {UseExecPath, UseSkipPath} instead of {false, true}

chapuni committed:
Catch up the merge

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Eliminate (rewind) "refactor" changes

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/condop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

chapuni committed:
Fix wrong merge resolutions

chapuni committed:
Reorganize CoverageMapping::SingleByteCoverage

chapuni committed:
Prune commented-out line

chapuni committed:
[Coverage] Make `MCDCRecord::Folded` as `[false/true]` with BitVector. NFC.

chapuni committed:
[Coverage] MCDC: Move `findIndependencePairs` into `MCDCRecord`

chapuni committed:
Merge branches 'users/chapuni/cov/merge/mcdcfold' and 'users/chapuni/cov/merge/mov_ind' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
[Coverage] Sort `MCDCRecord::ExecVectors` order by Bitmap index

chapuni committed:
llvm-cov: Refactor CoverageSummaryInfo. NFC.

chapuni committed:
Merge remote-tracking branches 'origin/users/chapuni/cov/single/refactor' and 'origin/users/chapuni/cov/single/unify' into users/chapuni/cov/merge/forfile-base

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/refactor

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test
	llvm/test/tools/llvm-cov/showLineExecutionCounts.test
	llvm/tools/llvm-cov/CodeCoverage.cpp
	llvm/tools/llvm-cov/SourceCoverageView.h

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/refactor

chapuni committed:
Merge branch 'users/chapuni/cov/single/refactor' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/single/refactor' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/trunk

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/summaryinfo

chapuni committed:
Merge branch 'users/chapuni/cov/merge/summaryinfo' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/binop-base

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/condop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branches 'users/chapuni/cov/single/switch' and 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Add a test

chapuni committed:
Prune redundant logic

chapuni committed:
Merge branch 'users/chapuni/cov/single/replace' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount-base' into users/chapuni/cov/single/nextcount

chapuni committed:
Expand RHS of MapToExpand. This will prevent recursion.

chapuni committed:
Append an explanation in the comment

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/binop-base

chapuni committed:
Update test

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mov_ind

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/single/unify' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mcdcsort-base' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mov_ind' into users/chapuni/cov/single/unify

chapuni committed:
Revert "Merge branch 'users/chapuni/cov/single/unify' into users/chapuni/cov/merge/mcdcsort-base"

This reverts commit c33e8987289f2775792977ee279c4e458f8091d5, reversing
changes made to ad6726dbfac5074c856f41adadb3b5bc14580311.

chapuni committed:
Rewind switch DefaultCase. (to #113112)

chapuni committed:
Enable addCounters

chapuni committed:
Get rid of structual bindings

chapuni committed:
Flatten with getBranchCounterPair(SkipCntForOld)

chapuni committed:
Merge branch 'users/chapuni/cov/single/getpair' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount-base' into users/chapuni/cov/single/nextcount

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

chapuni committed:
Resurrect merge conflict

chapuni committed:
Revert "Rewind switch DefaultCase. (to #113112)"

This reverts commit d854fb123be0449e83093f11e9f826c9baa3b766.

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

chapuni committed:
Dissolve assert

chapuni committed:
Dissolve assert

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/binop-base

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branches 'users/chapuni/cov/single/loop', 'users/chapuni/cov/single/condop' and 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Update comments

chapuni committed:
Decorate the mock

chapuni committed:
Dissolve the old behavior

chapuni committed:
Will be pruned after the migration of SingleByte.

chapuni committed:
Add comments

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/replace

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/getpair

chapuni committed:
Merge branch 'users/chapuni/cov/single/replace' into users/chapuni/cov/single/getpair

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/pair

chapuni committed:
Merge branch 'users/chapuni/cov/single/getpair' into users/chapuni/cov/single/pair

chapuni committed:
Merge branch 'users/chapuni/cov/single/pair' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount-base' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

chapuni committed:
Merge branches 'users/chapuni/cov/single/loop', 'users/chapuni/cov/single/condop' and 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
[MC/DC] Update CoverageMapping tests

To resolve the error, rename mcdc-error-nests.cpp -> mcdc-nested-expr.cpp at first.

- `func_condop`
  A corner case that contains close decisions.
- `func_expect`
  Uses `__builtin_expect`. (#124565)
- `func_lnot`
  Contains logical not(s) `!` among MC/DC binary operators. (#124563)

mcdc-single-cond.cpp is for #95336.

chapuni committed:
[MC/DC] Make covmap tolerant of nested Decisions

CoverageMappingWriter reorders `Region`s by `endLoc DESC` to
prioritize wider `Decision` with the same `startLoc`.

In `llvm-cov`, tweak seeking Decisions by reversal order to find
smaller Decision first.

chapuni committed:
[MC/DC] Refactor MCDC::State::Decision. NFC.

Introduce `ID` and `InvalidID`. Then `DecisionByStmt` can have three
states.

* Not assigned if the Stmt(Expr) doesn't exist.
* When `DecisionByStmt[Expr]` exists:
  * Invalid and should be ignored if `ID == Invalid`.
  * Valid if `ID != Invalid`. Other member will be filled in the
    Mapper.

chapuni committed:
[MC/DC] Refactor MCDCCoverageBuilder. NFC.

- Get rid of the old `DecisionStack` and dissolve it into push/pop
  `CurCondIDs` in `VisitBin`, since `VisitBin` is recursive.

- Introduce the new `DecisionStack` with `DecisionState` to handle the
  current `Decision` in nested `Decision`s.
  - The stack has the sentinel that has `DecisionExpr = nullptr`.
  - Split out `checkDecisionRootOrPush` from `pushAndAssignIDs` for
    non-BinOp. It assigns `CondID` to `E` (instead of assignment LHS
    in `pushAndAssignIDs`).
  - The stack is manupilated at the top Decision operator in `VisitBin`.
    - The stack grows at the entrance of the Decision with the initial
      state.
    - In the same level in `VisitBin`, the stack is popped and the
      `Decision` record is emitted.
- Introduce `DecisionEndToSince` to sweep `MCDCBranch`es partially in
  `cancelDecision`.

chapuni committed:
[MC/DC] Prune MCDCLogOpStack and use CGF.isMCDCDecisionExpr

`MCDCLogOpStack` is used only for detection of the Decision root. It
can be detected with `MCDC::State::DecisionByStmt`.

chapuni committed:
[MC/DC] Create dedicated MCDCCondBitmapAddr for each Decision

MCDCCondBitmapAddr is moved from `CodeGenFunction` into `MCDCState`
and created for each Decision.

In `maybeCreateMCDCCondBitmap`, Allocate bitmaps for all valid
Decisions and emit them order by ID, to prevent nondeterminism.

chapuni committed:
[MC/DC] Handle __builtin_expect as if parenthses

Fixes #124565

chapuni committed:
[MC/DC] Enable usage of `!` among `&&` and `||`

In the current implementation, `!(a || b) && c` was not treated as one
Decision with three terms.

Fixes #124563

chapuni committed:
Update ReleaseNotes

chapuni committed:
isValid()

chapuni committed:
Remove mcdc-single-cond.cpp

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/tests

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Prune mcdc-single-cond.cpp

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Refactor with expansion walking

chapuni commented:
Refactored with expansion walker.

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/nextcount

Conflicts:
	clang/lib/CodeGen/CodeGenFunction.h

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/loop

Conflicts:
	clang/lib/CodeGen/CGStmt.cpp
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/condop

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/switch

chapuni committed:
Merge branches 'users/chapuni/cov/single/loop', 'users/chapuni/cov/single/condop' and 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branches 'users/chapuni/cov/single/binop' and 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/covmapdesc

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/logopstack

chapuni committed:
clangCodeGen: Follow #142155

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/covgen

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp (#172557)

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/bitmapaddr

Conflicts:
	clang/lib/CodeGen/CodeGenFunction.h (#142155)

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Update mcdc-export-json.test (in #159119)

chapuni committed:
-EnableSingleByteCoverage

chapuni committed:
Add comments

chapuni committed:
ReleaseNotes.rst

chapuni committed:
emitCounterSetOrIncrement: Simplify.

- s/`TheMap`/`TheCounterPair`/
- Change the type of `Counter` to `ValueOpt`.

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/single/unify' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mcdcsort' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/loop

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/condop

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/trunk

Conflicts:
	clang/docs/ReleaseNotes.rst

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
fix: LoopFalse should be used.

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/expect

Conflicts:
	clang/include/clang/AST/IgnoreExpr.h
	clang/lib/CodeGen/CodeGenFunction.cpp

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/covgen

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/covgen

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/logopstack

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/logopstack

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/logopstack' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/expect' into users/chapuni/mcdc/nest/covgen

evodius96 commented:
> @evodius96 How about this? I think this is not NFC but a simple change.
> 
> I tried this for long time locally with the Clang tree. I expect this (and this series) will be fine.

I agree, but I really want to take time to study the changes in `CoverageMapping.cpp` before signing off. I'm working on it!

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/covgen

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/covgen' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/bitmapaddr' into users/chapuni/mcdc/nest/covmapdesc

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/bitmapaddr' into users/chapuni/mcdc/nest/covmapdesc

chapuni committed:
[Coverage] Introduce the predicate `hasSkipCounter(S)`. NFC.

Also, rename s/getIsCounterPair/getExecSkipCounterExistence/.

chapuni committed:
Merge commit 'cead042fd7b57f7bb454d4ffe6f0413658a4b069' into users/chapuni/cov/single/loop

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge commit 'cead042fd7b57f7bb454d4ffe6f0413658a4b069' into users/chapuni/cov/single/if

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge commit 'cead042fd7b57f7bb454d4ffe6f0413658a4b069' into users/chapuni/cov/single/switch

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

Conflicts:
	clang/docs/ReleaseNotes.rst

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/bitmapaddr' into users/chapuni/cov/single/condop

Conflicts:
	clang/lib/CodeGen/CGExprScalar.cpp
	clang/lib/CodeGen/CodeGenPGO.cpp
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/condop' into users/chapuni/cov/single/execskipexists

chapuni committed:
hasSkipCounter

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/loop

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.proftext
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/if

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.proftext
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/switch

chapuni committed:
getSwitchImplicitDefaultCounterPair: Flatten expressions.

I believe the original form was already readable in this context,
but I rewrote it to match the requested style.

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/loop

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/switch

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.proftext
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/loop

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/if

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.proftext
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml
	llvm/test/tools/llvm-cov/Inputs/branch-c-general.c
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.proftext
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/if

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.proftext
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/trunk' into users/chapuni/cov/single/binop

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/mcdc/nest/covmapdesc

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/covmapdesc
